spec/models/admin_spec.rb

```diff
@@ -0,0 +1,105 @@
+require 'rails_helper'
+
+# Define Admin tests.
+RSpec.describe Admin, type: :model do
+  # Define all validation tests.
+  context 'validation tests' do
+    it 'should fail validation when username is empty' do
+      admin = Admin.new(
+        email: 'test@email.com',
+        password: 'password'
+      ).save
+      expect(admin).to eq(false)
```

Comment:
you should also test that it fails validation for the _right_ reason.
I would restructure a bit, something like...
```
admin = Admin.new(...)
expect(admin.save).to eq(false) # This also reads better.
expect(admin.errors).to include('the appropriate message)
```

Reply:
Apply this to all validation tests.